File: Projet_SHAITA_DRIOUCHE_INFO0503/serveur_JAVA/src/package_java/enums/TypeEnergie.java
package package_java.enums;

public enum TypeEnergie {
    PETROLE("Petrole"),
    ELECTRICITE("Electricite"),
    HYDROGENE("Hydrogene"),
    BIOMETHANE("Biomethane"),
    GAZNATUREL("Gaz naturel");

    private String nom;

    TypeEnergie(String nom) {
        this.nom = nom;
    }

    public String toString() {
        return nom;
    }

    public static TypeEnergie getValue(String s) {
        for (TypeEnergie c : TypeEnergie.values()) {
            if (c.nom.equals(s))
                return c;
        }
        return null;
    }

}

File: Projet_SHAITA_DRIOUCHE_INFO0503/serveur_PHP/package_php/enum/TypeEnergie.php
<?php

class TypeEnergie {
    // Attributs
    public static $PETROLE = null;
    public static $ELECTRICITE = null;
    public static $HYDROGENE = null;
    public static $BIOMETHANE = null;
    public static $GAZNATUREL = null;
    

    private $nom;

    // Constructeur
    public function __construct($nom){
        $this->nom = $nom;
        
    }

    public function toString() : String{ return $this->nom; }


}

TypeEnergie::$PETROLE = new TypeEnergie("Petrole");
TypeEnergie::$ELECTRICITE = new TypeEnergie("Electricite");
TypeEnergie::$HYDROGENE = new TypeEnergie("Hydrogene");
TypeEnergie::$BIOMETHANE = new TypeEnergie("Biomethane");
TypeEnergie::$GAZNATUREL = new TypeEnergie("Gaz naturel");
   

$tabTypeEnergie = array(TypeEnergie::$PETROLE, TypeEnergie::$ELECTRICITE, TypeEnergie::$HYDROGENE, TypeEnergie::$BIOMETHANE, TypeEnergie::$GAZNATUREL);

?>